artifact-manifest/artifact-manifest.md

```diff
@@ -0,0 +1,462 @@
+    "size": 102
+  },
+  "blobs": [
+    {
+      "mediaType": "application/vnd.cncf.notary.v2.json",
+      "digest": "sha256:9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f0",
+      "size": 32654
+    }
+  ],
+  "dependencies": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1.config.json",
+      "digest": "sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b",
+      "size": 16724
+    }
+  ]
+}
+```
+
+#### OCI-Registry CLI
+
+To copy the above image and the associated signatures, a new `oci-reg` cli is proposed. The oci-reg cli is an independent tool that demonstrates the value of these collections, providing a unified standard means for working within and across different OCI compliant registry implementations.
+
+The following command would copy the `mysql:8` image from docker hub to the acme-rockets registry. The CLI could be run within the source or target cloud. 
+
+```bash
+oci-reg copy \
+  --source hub.docker.io/mysql:8 \
+  --target registry.acme-rockets.io/base-artifacts/mysql:8
+```
+
+The `oci-reg copy` command would:
+
+- assure the manifest and layer/blob digests remain the same
+- copy any artifacts that are dependent on the source artifact-manifest, persisting them in the target registry.
+
+### Reference Artifacts
+
+There are a set of artifact types that declare references to other artifacts that may, or may not be stored in the same registry. The reference is important to note, indicating copying to be capable between environments, as well as generalized validation scenarios.
+
+#### Helm Reference
+
+![mysql image copy](./media/wordpress-helm-chart-copy.svg)
+
+In the above scenario, a helm chart is copied from a public registry to the ACME Rockets registry.  The `wordpress-chart:v5` is represented as an `application/vnd.oci.artifact.manifest.v1+json`. The `wordpress-chart:v5` helm chart references the `wordpress:v5` image and the `mysql:8` image. All three artifacts have signatures attesting to their authenticity.
+
+As the copy is initiated, the `oci.artifact.manifest` of the `wordpress-chart:v5` is evaluated. As the chart references the same version (digest) of the `mysql:8` image already in theACME Rockets registry, the copy skips duplicating the content and moves to copying the `wordpress:v5` image, the `wordpress-chart:v5` and their associated signatures.
+
+In this case, the images referenced in the chart existed in the source registry. However, this may not be the case in all scenarios. A helm chart may be acquired from a different location, referencing images from docker hub. This deferred validation is left to client tools, enabling common package management resolution scenarios.
+
+To support the loose references between artifacts, a `references` collection is added to the `oci.artifact.manifest`:
+
+**A `wordpress-chart:v5` Helm Chart example:**
+
+```json
+{
+  "schemaVersion": 2,
```

Comment:
Everything here and below will need to be `"schemaVersion": 3`, I think, as these are breaking changes?

Reply:
This is actually a new schema as we're introducing a new `oci.artifact.manifest` that would rev independently from `oci.image` and `oci.index`.

Since the `oci.artifact.manifest` is versioned in the `mediaType=application/oci.artifact.manifest.v1`, I've removed the schemaVersion element.

Reply:
No, that's not how `schemaVersion` works.

Reply:
I've added `"schemaVersion": 1` back in as the first scheamVersion for `oci.artifact.manifest`